### ISSUE TITLE
[Enhancement]: Match New Authors feature

### ISSUE THREAD

**Issue by raviwarrier** [advplyr/audiobookshelf#1789]
### Describe the feature/enhancement

In the authors' page, there is only one possible action: "Match All Authors". This action starts working on all authors, irrespective of whether they have info or not. See image below:
![image](https://github.com/advplyr/audiobookshelf/assets/42061703/5974c518-4647-4a05-bf89-6b8dd0574330)

In addition to "match all authors", also have "match new authors" and/or "match unmatched authors" (phrasing can be better) to work only on those authors that are newly added or those who don't have info matched. 

This should save processing time, audible limits and overall time for library updates.

**Comment by catduckgnaf** [comment]
I agree, can it simply do a "match on new author" like when a new author is found, try a quickmatch?

**Comment by advplyr** [comment]
What would the criteria be for a new or unmatched author? It doesn't have an ASIN? Image? Description?

**Comment by raviwarrier** [comment]
> What would the criteria be for a new or unmatched author? It doesn't have an ASIN? Image? Description?

It would, I presume, be the same as what you do for "Match All Authors", but "added date > last matching done date", or "has description = false"

This depends on whether you do have a way of tracking 'last matching done date', or maintain an 'added date' field for authors.

the easiest way would be to see if the book associated with that author has been added recently, and if yes, check if the author record has details, and if not, then do a quick match. since you do maintain "added date" for the book.

If book new > if author details missing > quick match.

if the book is old, one can presume that the author has been updated: searched, matched and updated OR searched, not found, not updated - are the only two possible outcomes.

**Comment by catduckgnaf** [comment]
> > What would the criteria be for a new or unmatched author? It doesn't have an ASIN? Image? Description?
> 
> It would, I presume, be the same as what you do for "Match All Authors", but "added date > last matching done date", or "has description = false"
> 
> This depends on whether you do have a way of tracking 'last matching done date', or maintain an 'added date' field for authors.
> 
> the easiest way would be to see if the book associated with that author has been added recently, and if yes, check if the author record has details, and if not, then do a quick match. since you do maintain "added date" for the book.
> 
> If book new > if author details missing > quick match.
> 
> if the book is old, one can presume that the author has been updated: searched, matched and updated OR searched, not found, not updated - are the only two possible outcomes.

I agree with this approach, and as far as details, I think description and/or image. I would avoid looking at ASIN, or at least exclusively. I have added many authors manually, and I may not add or have their ASIN.
